test/e2e/registration_service_test.go

```diff
@@ -39,15 +45,417 @@ func (s *registrationServiceTestSuite) TestLandingPageReachable() {
+
+		resp, err := client.Do(req)
+		require.NoError(s.T(), err)
+
+		defer close(s.T(), resp)
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+
+		assert.Equal(s.T(), http.StatusNotFound, resp.StatusCode)
+	})
+}
+
+func (s *registrationServiceTestSuite) TestSignupOK() {
+	// Get valid generated token for e2e tests. IAT claim is overriden
+	// to avoid token used before issued error.
+	identity0 := authsupport.NewIdentity()
+	emailClaim0 := authsupport.WithEmailClaim(uuid.NewV4().String() + "@email.tld")
+	iatClaim0 := authsupport.WithIATClaim(time.Now().Add(-60 * time.Second))
+	token0, err := authsupport.GenerateSignedE2ETestToken(*identity0, emailClaim0, iatClaim0)
+	require.NoError(s.T(), err)
+
+	// Call signup endpoint with an valid token.
+	req, err := http.NewRequest("POST", s.route+"/api/v1/signup", nil)
+	require.NoError(s.T(), err)
+	req.Header.Set("Authorization", "Bearer "+token0)
+	req.Header.Set("content-type", "application/json")
+	client := getClient()
+
+	resp, err := client.Do(req)
+	require.NoError(s.T(), err)
+
+	defer close(s.T(), resp)
+
+	body, err := ioutil.ReadAll(resp.Body)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), body)
+	assert.Equal(s.T(), http.StatusAccepted, resp.StatusCode)
+
+	// Attempt to create same usersignup by calling post signup with same token
+	resp, err = client.Do(req)
+	require.NoError(s.T(), err)
+
+	defer close(s.T(), resp)
+
+	body, err = ioutil.ReadAll(resp.Body)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), body)
+
+	mp := make(map[string]interface{})
+	err = json.Unmarshal([]byte(body), &mp)
+	require.NoError(s.T(), err)
+
+	assert.Equal(s.T(), "usersignups.toolchain.dev.openshift.com \""+identity0.ID.String()+"\" already exists", mp["message"])
+	assert.Equal(s.T(), "error creating UserSignup resource", mp["details"])
+	assert.Equal(s.T(), http.StatusInternalServerError, resp.StatusCode)
+
+	// Call get signup endpoint with an valid token.
+	req, err = http.NewRequest("GET", s.route+"/api/v1/signup", nil)
+	require.NoError(s.T(), err)
+	req.Header.Set("Authorization", "Bearer "+token0)
+	req.Header.Set("content-type", "application/json")
+
+	resp, err = client.Do(req)
+	require.NoError(s.T(), err)
+
+	defer close(s.T(), resp)
+
+	body, err = ioutil.ReadAll(resp.Body)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), body)
+
+	mp = make(map[string]interface{})
+	err = json.Unmarshal([]byte(body), &mp)
+	require.NoError(s.T(), err)
+
+	mpStatus, ok := mp["status"].(map[string]interface{})
+	assert.True(s.T(), ok)
+
+	assert.Nil(s.T(), mp["CompliantUsername"])
+	assert.Equal(s.T(), identity0.Username, mp["username"])
+	require.IsType(s.T(), false, mpStatus["ready"])
+	assert.False(s.T(), mpStatus["ready"].(bool))
+	assert.Equal(s.T(), "PendingApproval", mpStatus["reason"])
+
+	userSignup, err := s.awaitility.Host().WaitForUserSignup(identity0.ID.String(), wait.UntilUserSignupHasConditions(pendingApproval()...))
```

Comment:
Let's maybe move this wait for PendingApproval before we check the status in GET /signup in line 382?